### PR TITLE
Add Market research checkbox

### PR DESCRIPTION
**Pull request by Calum-Campbell**
## What does this change?
- Adds a market research opt-out checkbox to the email and consents pages
- Bumps the Identity library to get new wording for this consent deployed in https://github.com/guardian/identity/pull/1179

## Screenshots
image

### CODE REVIEW

File: identity/app/form/PrivacyMapping.scala [hunk 1]
```diff
@@ -122,15 +122,21 @@ object PrivacyFormData extends SafeLogging {
     * Filter out any invalid consents that are still lingering in Mongo DB.
     *
     * For example, if consent id is renamed, then some users might still have the old consent.
-    *
```
Comment (walaura) [inline identity/app/form/PrivacyMapping.scala]: qq

File: identity/app/views/profile/nonElectronicOptOut.scala.html [hunk 2]
```diff
@@ -27,6 +27,19 @@
...
+                        also contact you by phone.</p>
+                    <li>
+                    @fragments.consentSwitch(
+                        consentField,
+                        boldTitle = false,
+                        titleProvider = wording => wording.description.getOrElse(""),
+                        descriptionProvider = _ => None
+                    )(messages)
+                    </li>
+                }
```
Comment (walaura) [inline identity/app/views/profile/nonElectronicOptOut.scala.html]: html looks good!

**Review by walaura — APPROVED** [review]
can't comment on the scala bits

File: identity/app/views/profile/nonElectronicOptOut.scala.html [hunk 3]
```diff
@@ -27,6 +27,19 @@
                         )(messages)
                     </li>
                 }
+                @if(isMarketResearch(consentField, user)) {
+                    <p class="form__heading">Market Research</p>
+                    <p class="identity-title-explainer">From time to time we may contact you for market research purposes inviting you to complete a survey, or take part in a group discussion. Normally, this invitation would be sent via email, but we may
+                        also contact you by phone.</p>
```
Comment (jfsoul) [inline identity/app/views/profile/nonElectronicOptOut.scala.html]: why's there a line break here?

**Review by jfsoul — APPROVED** [review]

**Pull request merged**